src/pages/candidate-details/candidate-details.model.ts

```diff
@@ -0,0 +1,26 @@
+
+export type SlotTypeView = {
+  text: string,
```

Comment:
I'm not sure I get the point of this type. From reading the PR the SlotTypeView contains if the slot is single or double and if it should show the indicator for special needs.

Wouldn't a simpler solution be to set slotType in the Details type to be a string and in the candidate-details.html file the ng if for the indicator could be slotType === 'Single'

You could put some more safety by only allowing either single or double via a enum or something else.

Reply:
The component's html file should have one reason to change. And that is displaying the data differently. So that's the reason I moved the logic out to the selector, which prepares the data for the html template basically.

And at the moment we have only 2 types, double slot and single slot. But today we already had a discussion about how many different slot types exists. And we need to implement all of them so there might be other icons.

This setup gives us flexibility for these changes.